Tving-Clone-Project/Tving-Clone-Project/ViewController/MainViewController.swift

```diff
@@ -0,0 +1,425 @@
+                                        withReuseIdentifier: HeaderView.identifier)
+    }
+    
+    private func setLayout() {
+        [
+            scrollView,
+            bufferView
+        ].forEach { self.view.addSubview($0) }
+        
+        scrollView.addSubview(contentView)
+        posterScrollView.addSubview(posterContentView)
+        
+        [
+            mainPosterImageView,
+            tvingTopLogoImageView,
+            rightTopButtonStackView,
+            tabControlCollectionView,
+            posterScrollView,
+            pageControl,
+            mainCollectionView
+        ].forEach { contentView.addSubview($0) }
+        [
+            posterTitleLabel,
+            posterDetailLabel
+        ].forEach { posterContentView.addSubview($0) }
+        
+        bufferView.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.width.equalToSuperview()
+            $0.height.equalTo(80)
+        }
+        scrollView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        contentView.snp.makeConstraints {
+            $0.edges.width.equalToSuperview()
+            $0.height.equalTo(1000)
+        }
+        
+        mainPosterImageView.snp.makeConstraints {
+            $0.top.leading.trailing.equalToSuperview()
+            $0.height.equalTo(posterHeight)
+        }
+        tvingTopLogoImageView.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(5)
+            $0.leading.equalToSuperview().inset(20)
+        }
+        rightTopButtonStackView.snp.makeConstraints {
+            $0.centerY.equalTo(tvingTopLogoImageView)
+            $0.trailing.equalToSuperview().inset(20)
+        }
+        tabControlCollectionView.snp.makeConstraints {
+            $0.top.equalTo(bufferView.snp.bottom)
+            $0.horizontalEdges.equalToSuperview()
+            $0.height.equalTo(40)
+        }
+        posterScrollView.snp.makeConstraints {
+            $0.top.leading.trailing.equalToSuperview()
+            $0.height.equalTo(posterHeight)
+        }
+        posterContentView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+            $0.width.greaterThanOrEqualTo(Int(deviceWidth) * posterImages.count)
+        }
+        posterTitleLabel.snp.makeConstraints {
+            $0.leading.equalToSuperview().inset(20)
+            $0.bottom.equalTo(mainPosterImageView).offset(-70)
+        }
+        posterDetailLabel.snp.makeConstraints {
+            $0.leading.equalTo(posterTitleLabel)
+            $0.top.equalTo(posterTitleLabel.snp.bottom).offset(15)
+        }
+        pageControl.snp.makeConstraints {
+            $0.top.equalTo(mainPosterImageView.snp.bottom).offset(45)
+            $0.leading.equalToSuperview().offset(-23)
+        }
+        mainCollectionView.snp.makeConstraints {
+            $0.top.equalTo(pageControl.snp.bottom).offset(25)
+            $0.leading.trailing.bottom.equalToSuperview().inset(15)
```

Comment:
```suggestion
            $0.leading.trailing.equalToSuperview().inset(15)
            $0.bottom.equalTo(contentView.snp.bottom).inset(15)
```
알고 계실수도 있지만 위에 contentView의 높이를 고정하지 않고 이런 식으로 코드를 써주시는 방법도 있습니다! 안에 들어가는 컴포넌트들의 높이에 따라 유동적으로 contentView의 높이가 바뀔 수 있어서 저는 이 방법을 많이 씁니다!

Reply:
앗 이미 그렇게 돼있습니다! 아마 커밋 단위로 리뷰하셔서 예전 버전을 보신 거 아닌가 추측해봅니다!